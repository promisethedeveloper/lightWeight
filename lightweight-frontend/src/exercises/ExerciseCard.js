import React from "react";
import { Link } from "react-router-dom";

import "./ExerciseCard.css";

/** Show limited information about an exercise
 *
 * Is rendered by ExerciseList to show a "card" for each exercise.
 *
 * ExerciseList -> ExerciseCard
 */

const ExerciseCard = ({
	id: exercise_id,
	name,
	bodyPart,
	equipment,
	target,
	gifUrl,
	workout_Id,
}) => {
	return (
		<Link
			className="ExerciseCard card"
			to={`/exercises/${workout_Id}/${exercise_id}/${name}`}
		>
			<div className="card-body">
				<p className="card-title">Name: {name}</p>
				<img src={gifUrl} alt={name} className="float-right ml-5" />
				<p>Body part: {bodyPart}</p>
				<p>Target muscle: {target}</p>
				<p>Equipment: {equipment}</p>
			</div>
		</Link>
	);
};

export default ExerciseCard;
