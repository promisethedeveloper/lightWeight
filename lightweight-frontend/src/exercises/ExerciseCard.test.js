import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import ExerciseCard from "./ExerciseCard";

/** Smoke test */
it("Should render without crashing", () => {
	render(
		<MemoryRouter>
			<ExerciseCard />
		</MemoryRouter>
	);
});

/** Snapshot test */
it("Should match snapshots", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<ExerciseCard />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it("should load an exercise card", () => {
	const { getByText } = render(
		<MemoryRouter>
			<ExerciseCard
				name={"Bench press"}
				bodyPart={"Chest"}
				equipment={"Barbell"}
				target={"Pectoral"}
			/>
		</MemoryRouter>
	);
	expect(getByText("Name: Bench press")).toBeInTheDocument();
	expect(getByText("Body part: Chest")).toBeInTheDocument();
	expect(getByText("Equipment: Barbell")).toBeInTheDocument();
	expect(getByText("Target muscle: Pectoral")).toBeInTheDocument();
});
