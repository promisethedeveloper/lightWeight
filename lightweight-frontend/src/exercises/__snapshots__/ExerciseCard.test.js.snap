// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshots 1`] = `
<DocumentFragment>
  <a
    class="ExerciseCard card"
    href="/exercises/undefined/undefined/undefined"
  >
    <div
      class="card-body"
    >
      <p
        class="card-title"
      >
        Name: 
      </p>
      <img
        class="float-right ml-5"
      />
      <p>
        Body part: 
      </p>
      <p>
        Target muscle: 
      </p>
      <p>
        Equipment: 
      </p>
    </div>
  </a>
</DocumentFragment>
`;
